### PR TITLE
New version 1.4.0 - PHP 8.1 compatibility, Compatibility with WooComm…

### PR DESCRIPTION
**Pull request by jineshpv**
- PHP 8.1 compatibility
- Compatibility with WooCommerce 7.7
- Updated the API version to 69

### CODE REVIEW

File: templates/checkout/hostedsession.php [hunk 1]
```diff
@@ -77,7 +77,7 @@ function hsErrorsMap() {
     }
 
     function mpgsPayWithSelectedInstrument() {
-        var selected = document.querySelectorAll('[name=wc-mpgs_gateway-payment-token]:checked')[0];
+        var selected = document.querySelectorAll('[name=wc-mpgs_gateway-payment-token]:checked')[0]; console.log(selected);
```
Comment (mdhilwan) [inline templates/checkout/hostedsession.php:80]: console.log?

File: includes/class-gateway.php [hunk 2]
```diff
@@ -26,15 +26,11 @@ class Mastercard_Gateway extends WC_Payment_Gateway {
 
 	const ID = 'mpgs_gateway';
 
-	const MPGS_API_VERSION = 'version/63';
-	const MPGS_API_VERSION_NUM = '63';
-
-	const MPGS_LEGACY_API_VERSION = 'version/61';
-	const MPGS_LEGACY_API_VERSION_NUM = '61';
+	const MPGS_API_VERSION = 'version/69';
+	const MPGS_API_VERSION_NUM = '69';
```
Comment (mdhilwan) [inline includes/class-gateway.php:30]: 63 or 69?

Comment (mdhilwan) [reply includes/class-gateway.php:30]: i see a number of pr pointing to 63 instead of 69

File: templates/checkout/hostedsession.php (hunk 1 above)
Comment (jineshpv) [reply templates/checkout/hostedsession.php]: @mdhilwan The console.log function was taken out from hostedsession.php.

File: includes/class-gateway.php (hunk 2 above)
Comment (jineshpv) [reply includes/class-gateway.php:30]: @mdhilwan We have updated the MPGS API version to 69 (the latest version).

File: includes/class-gateway-service.php [hunk 3]
```diff
@@ -526,7 +523,7 @@ public function createCheckoutSession(
 
 	/**
 	 * Request to add or update request fields contained in the session.
-	 * PUT    https://test-gateway.mastercard.com/api/rest/version/58/merchant/{merchantId}/session/{sessionId}
+	 * PUT    https://test-gateway.mastercard.com/api/rest/version/63/merchant/{merchantId}/session/{sessionId}
```
Comment (mdhilwan) [inline includes/class-gateway-service.php:526]: But I do see v63 here. But v69 being used in the API calls. Is this done on purpose?

Comment (jineshpv) [reply includes/class-gateway-service.php:526]: @mdhilwan That's the comment section. The comment section does not affect the code base; its purpose is to describe how the functionality operates.

Comment (mdhilwan) [reply includes/class-gateway-service.php:526]: Orite cool. Just checking

**Review by mdhilwan — APPROVED** [review]
Approved

**Pull request merged**